src/main/java/programminglife/model/drawing/DrawableSegment.java

```diff
@@ -0,0 +1,241 @@
+     * @return children {@link Collection} are the direct children of the node {@link DrawableSegment}.
+     */
+    @Override
+    public Collection<Integer> getChildren() {
+        return children;
+    }
+
+    /**
+     * Get all the parents of the node {@link DrawableSegment}.
+     * @return parent {@link Collection} are the direct parents of the node {@link DrawableSegment}.
+     **/
+    public Collection<Integer> getParents() {
+        return parents;
+    }
+
+    /**
+     * Replace a child node with a dummy node.
+     * @param oldChild The {@link DrawableSegment} to replace.
+     * @param newChild The {@link DrawableSegment} to replace with.
+     */
+    void replaceChild(DrawableNode oldChild, DrawableNode newChild) {
+        if (!this.children.remove(oldChild.getIdentifier())) {
+            throw new NoSuchElementException("The node to be replaced is not a child of this node.");
+        }
+        this.children.add(newChild.getIdentifier());
+    }
+
+    @Override
+    public String details() {
+        return String.format("Sequence: %s%nGenomes: %s", this.getSequence(), this.getGraph().getGenomeNames(this.getGenomes()));
+    }
+
+    /**
+     * Replace a parent node with a dummy node.
+     * @param oldParent The {@link DrawableSegment} to replace.
+     * @param newParent The {@link DrawableSegment} to replace with.
+     */
+    void replaceParent(DrawableNode oldParent, DrawableNode newParent) {
+        if (!this.parents.remove(oldParent.getIdentifier())) {
+            throw new NoSuchElementException(
+                    String.format("The node to be replaced (%d) is not a parent of this node (%d).",
+                            oldParent.getIdentifier(), this.getIdentifier()));
+        }
+        this.parents.add(newParent.getIdentifier());
+    }
+
+    public int getIntX() {
+        return (int) Math.ceil(this.getX());
+    }
+
+    public int getIntY() {
+        return (int) Math.ceil(this.getY());
+    }
+
+    public int getIntWidth() {
+        return (int) Math.ceil(this.getWidth());
+    }
+
+    public int getIntHeight() {
+        return (int) Math.ceil(this.getHeight());
+    }
+
+    /**
+     * Set the size {@link XYCoordinate} of the Segment.
+     * @param size The {@link XYCoordinate} representing the size
+     */
+    void setSize(XYCoordinate size) {
+        this.setWidth(size.getX());
+        this.setHeight(size.getY());
+        this.setDrawDimensionsUpToDate(true);
+    }
+
+    /**
+     * Set an {@link XYCoordinate} representing the location of the Segment.
+     */
+    void setLocation(int x, int y) {
+        this.setX(x);
+        this.setY(y);
+    }
+
+    /**
+     * getter for the width coordinate.
+     * @return XYCoordinate.
+     */
+    public XYCoordinate getWidthCoordinate() {
+        if (!this.isDrawDimensionsUpToDate()) {
+            setDrawDimensions();
+        }
+        return new XYCoordinate((int) this.getWidth(), 0);
```

Comment:
If `y = 0`, then why return an XYCoordinate instead of an `int` / `double`?
Else, why not also return the height and rename it to getSize?